lib/rust/ensogl/core/src/debug/stats.rs

```diff
@@ -84,7 +126,30 @@ gen_stats! {
     shader_compile_count : usize,
 }
 
+
+// === StatsData methods ===
+
 impl StatsData {
+    fn begin_frame(&mut self, time: f64) {
+        self.frame_counter += 1;
+
+        if self.frame_begin_time > 0.0 {
+            let end_time = time;
+            self.fps = 1000.0 / (end_time - self.frame_begin_time);
+        }
+        self.frame_begin_time = time;
+    }
+
+    fn end_frame(&mut self, time: f64) {
+        self.frame_time = time - self.frame_begin_time;
+
+        if cfg!(target_arch = "wasm32") {
```

Comment:
We are not using such checks nowhere in the code. Why it is used here?

Reply:
It's used here because `wasm_bindgen::memory()` panics in non-wasm_bindgen code, making it impossible to use `end_frame` in regular tests, as long as it contains WASM memory calculation. Alternatives I considered:

 1. Drop the check and make the tests be `#[wasm_bindgen_test]` instead of `#[test]` - however, this makes running the tests harder and slows down development iteration time.
 2. Use a compilation guard attribute on a block of code, instead of the `if cfg!(...)`:

        #[cfg(target = "wasm32")]
        {
            let memory: ...
            ...
        }

 3. Move the calculation of `wasm_memory_usage` outside of `end_frame()`, to a separate new method, and call this method externally from somewhere - though I didn't have a good idea what would be a good place to call it from.

Would you suggest I pick any other of the alternatives proposed above, or add a comment in code explaining the check, or do something else still?

Reply:
There is yet another alternative – @vitvakatu is right now working on allowing all wasp-related functions to return something (like 0) instead of panicking when compiled to a native code. We need exactly this thing in so many places, that handling it here differently than in other places doesn't make much sense. However, as Ilya's PR is not ready yet, let's leave it this way with an explanation and TODO pointing to Ilya's task explaining that we should move it to the new solution as soon as it will be available. How does it sound to you?

Reply:
I love this idea; just added the TODO, thanks! ❤️